src/Groups/matrices/transform_form.jl

```diff
@@ -0,0 +1,412 @@
+# TODO: in this file are used many methods TEMPORARILY defined in files matrix_manipulation.jl and stuff_field_gen.jl
+# once methods in those files will be deleted / replaced / modified, this file need to be modified too
+
+
+
+import AbstractAlgebra: Field
+
+export
+    iscongruent
+
+###############################################################################################################
+
+# Algorithm
+
+###############################################################################################################
+
+# The functions find_radical, block_anisotropic_elim and block_herm_elim are based on the article
+# James B. Wilson, Optimal Algorithms of Gram-Schmidt type, Linear Algebra and its Applications 438 (2013) 4573-4583
+
+# returns x,y such that ax^2+by^2 = c
+# at the moment, it simply search for x such that (c-ax^2)/b is a square
+# TODO: is there a faster way to find x and y?
+# TODO: it would be better if this is deterministic. This depends on gen(F) and issquare(F).
```

Comment:
@fieker is there perhaps already a suitable function for this in e.g. Hecke?

Reply:
On a different note: above you do h = F(1) outside the loop, I guess to define it so it can be assigned inside. So there are some "better" options:
- `local h` which will not actually waste a new element `F(1)`
- instead of the break, do the testing and return inside the loop
- use (try to use) findfirst to get the index of the first non-zero entry

Max's question:
- I don't know, Simon and Tommy are doing quadratic stuff
- there is no reason to involve gen(F) to do rand(F)
- to make it "deterministic" use a local RNG, every second x should work
- issquare is a function that Bill loves. He might be able to answer that question better

Reply:
Was this code written before one could iterate over finite fields? If so, it should just be `for x in F ... `.